spring-cloud-kubernetes-leader/src/main/java/org/springframework/cloud/kubernetes/leader/KubernetesHelper.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2018 to the original authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.leader;
+
+import java.util.Map;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+/**
+ * @author <a href="mailto:gytis@redhat.com">Gytis Trikleris</a>
+ */
+public class KubernetesHelper {
+
+	private static final String PROVIDER_KEY = "provider";
+
+	private static final String PROVIDER = "spring-cloud-kubernetes";
+
+	private static final String KIND_KEY = "kind";
+
+	public static final String KIND = "leaders";
+
+	private final LeaderProperties leaderProperties;
+
+	private final KubernetesClient kubernetesClient;
+
+	public KubernetesHelper(LeaderProperties leaderProperties, KubernetesClient kubernetesClient) {
+		this.leaderProperties = leaderProperties;
+		this.kubernetesClient = kubernetesClient;
+	}
+
+	public boolean isPodAlive(String id) {
```

Comment:
This just checks if the pod exists, it doesn't do any status check. If its essential for the pod to be alive, I'd suggest we include the check. If not let's just change the name of the method to `podExists`.

Reply:
It is possible for a pod to become a leader during the application boot-up, so having a status check might interfere with this. I'll rename to `podExists`